src/test/java/uk/ac/cam/cl/dtg/isaac/api/managers/GameManagerTest.java

```diff
@@ -92,4 +95,120 @@ public void getNextQuestionsForFilter_appliesExclusionFilterForDeprecatedQuestio
     assertEquals(deprecatedFilter.getOperator(), Constants.BooleanOperator.NOT);
     assertEquals(deprecatedFilter.getValues(), Collections.singletonList("true"));
   }
+
+  @Test
+  public void generateRandomQuestions_returnsCorrectNumberOfQuestions() throws ContentManagerException {
+
+    // Arrange
+    int limit = 5;
+
+    var gameManager = new GameManager(
+        this.dummyContentManager,
+        this.dummyGameboardPersistenceManager,
+        this.dummyMapper,
+        this.dummyQuestionManager,
+        "latest"
+    );
+
+    List<IsaacQuestionPageDTO> questions = new ArrayList<>();
+    for (int i = 0; i < limit; i++) {
+      IsaacQuestionPageDTO question = createMock(IsaacQuestionPageDTO.class);
+      question.setSupersededBy(null);
```

Comment:
I don't think this is doing what you intended it to do. `question` is a mock, so it does not behave like that, I don't think.